src/gsy_e/models/strategy/energy_parameters/energy_params_eb.py

```diff
@@ -1,16 +1,157 @@
+from typing import Dict, Optional
 
 import pendulum
+from gsy_framework.utils import convert_kW_to_kWh
+
+from gsy_e.gsy_e_core.enums import AvailableMarketTypes
+from gsy_e.models.state import LoadState
+from gsy_e.models.strategy.energy_parameters.standard_profile_builders import StandardProfileParser
 
 logger = getLogger(__name__)
 
 
+class StandardProfileException(Exception):
+    """Exception for the StandardProfile."""
+
+
+class LoadSSPEnergyParameters:
+    """Energy parameters of the load strategy for the forward markets."""
+
+    _STANDARD_SOLAR_PROFILE = StandardProfileParser.parse()
+
+    def __init__(self, capacity_kW):
+        self.state = LoadState()
+        self.capacity_kW: float = capacity_kW
+
+        self._area = None
+        self._profile_generator: Optional[ForwardTradeProfileGenerator] = None
+
+    def serialize(self):
+        """Return dict with the current energy parameter values."""
+        return {
+            "capacity_kW": self.capacity_kW
+        }
+
+    def event_activate_energy(self, area):
+        """Update energy requirement upon the activation event."""
+        self._area = area
+        self._profile_generator = ForwardTradeProfileGenerator(
+            peak_kWh=convert_kW_to_kWh(self.capacity_kW, self._area.config.slot_length))
+
+    def event_traded_energy(self, energy_kWh, market_slot: pendulum.DateTime, product_type):
+        """
+        When a trade happens, we want to split the energy through the entire year following a
+        standard solar profile.
+        """
+        assert self._profile_generator is not None
+        # Create a new profile that spreads the trade energy across multiple slots. The values
+        # of this new profile are obtained by scaling the values of the standard solar profile
+        return self._profile_generator.generate_trade_profile(
+            energy_kWh=energy_kWh,
+            market_slot=market_slot,
+            product_type=product_type)
+
+
+class ForwardTradeProfileGenerator:
+    """
+    Class to generate a new profile that spreads the trade energy across multiple market slots.
+
+    The values of this profile are obtained by scaling the values of the Standard Solar Profile.
+    """
+
+    _STANDARD_SOLAR_PROFILE = StandardProfileParser.parse()
+
+    def __init__(self, peak_kWh: float):
+        self._scaler = ProfileScaler(self._STANDARD_SOLAR_PROFILE)
+        self._scaled_capacity_profile = self._scaler.scale_by_peak(peak_kWh=peak_kWh)
+
+    def generate_trade_profile(
+            self, energy_kWh, market_slot, product_type):
+        """Generate a profile based on the market slot and the product type."""
+        profile_scaled_by_traded_energy = self._scaler.scale_by_peak(peak_kWh=energy_kWh)
+
+        # This subtraction is done _before_ expanding the slots to improve performance
+        residual_energy_profile = self._subtract_profiles(
+            self._scaled_capacity_profile, profile_scaled_by_traded_energy)
```

Comment:
Minor and can be skipped in order to avoid premature optimization, but it could be further optimized if we maintained the peak of the scaled_capacity_profile. That way, we would subtract only 2 floating point values, and then use only once the scale_by_peak method without any need for subtracting the 2 profiles. As mentioned, no need to deal with it now though but only if this becomes a problem

Reply:
I noticed that it looks like you commented on an outdated version of the code. I commented out these lines because I'm not dealing with the residual energy, but only with the traded energy.

Reply:
Thanks for pointing it out, I forgot to reply to this comment after I saw the later commits! Looks good as is!